Restore wallet throws MST error

Restoring wallet shows error: 
`[mobx-state-tree] Cannot modify 'WalletStore@/walletStore', the object is protected and can only be modified using an action.`

```
 (NOBRIDGE) ERROR  [WalletStore] Restoration error: [Error: [mobx-state-tree] Error while converting `null` to `(string | undefined)`:

    value `null` is not assignable to type: `(string | undefined)` (No type is applicable for the union).
    value `null` is not assignable to type: `(string | undefined)` (Value is not a string).
    value `null` is not assignable to type: `(string | undefined)` (Value is not a undefined).]
```

--- a/app/models/user/UserStore.ts
+++ b/app/models/user/UserStore.ts
@@ -1,6 +1,5 @@
 import { Instance, SnapshotIn, SnapshotOut, types } from "mobx-state-tree"
 import { withSetPropAction } from "../_helpers/withSetPropAction"
-import { isDevEnvironment } from "@/config"
 
 export const UserStoreModel = types
   .model("UserStore")
@@ -9,13 +8,6 @@ export const UserStoreModel = types
   })
   .actions(withSetPropAction)
   .actions((self) => ({
-    afterCreate() {
-      // Set example push token in dev environment
-      if (isDevEnvironment) {
-        self.pushToken = "example-push-token"
-      }
-    },
-
     setPushToken(token: string) {
       self.pushToken = token
     },
@@ -36,5 +28,5 @@ export interface UserStoreSnapshotIn extends SnapshotIn<typeof UserStoreModel> {
 
 export const createUserStoreDefaultModel = () =>
   UserStoreModel.create({
-    pushToken: isDevEnvironment ? "example-push-token" : "",
+    pushToken: "",
   })

--- a/app/models/wallet/WalletStore.ts
+++ b/app/models/wallet/WalletStore.ts
@@ -40,7 +40,7 @@ const WalletStoreModel = types
     setSetupComplete(complete: boolean) {
       self.setupComplete = complete
     },
-    setMnemonic(mnemonic: string) {
+    setMnemonic(mnemonic: string | undefined) {  // Updated to match types.maybe(types.string)
       self.mnemonic = mnemonic
     },
     setError(message: string | null) {

--- a/app/models/wallet/actions/disconnect.ts
+++ b/app/models/wallet/actions/disconnect.ts
@@ -8,8 +8,8 @@ export async function disconnect(store: IWalletStore) {
       await breezService.disconnect()
     }
     await SecureStorageService.deleteMnemonic()
-    store.isInitialized = false
-    store.mnemonic = null
+    store.setInitialized(false)
+    store.setMnemonic(undefined)
     store.setError(null)
   } catch (error) {
     console.error("[WalletStore] Disconnect error:", error)

--- a/app/models/wallet/actions/restoreWallet.ts
+++ b/app/models/wallet/actions/restoreWallet.ts
@@ -12,12 +12,12 @@ export async function restoreWallet(store: IWalletStore, mnemonic: string) {
     }
 
     // Reset the store state
-    store.isInitialized = false
-    store.balanceSat = 0
-    store.pendingSendSat = 0
-    store.pendingReceiveSat = 0
-    store.transactions.clear()
-    store.mnemonic = null
+    store.setInitialized(false)
+    store.setBalanceSat(0)
+    store.setPendingSendSat(0)
+    store.setPendingReceiveSat(0)
+    store.setTransactions([])
+    store.setMnemonic(null)
 
     // Validate and save mnemonic to secure storage
     const saved = await SecureStorageService.setMnemonic(mnemonic)
@@ -26,7 +26,7 @@ export async function restoreWallet(store: IWalletStore, mnemonic: string) {
     }
 
     // Set mnemonic in store
-    store.mnemonic = mnemonic
+    store.setMnemonic(mnemonic)
 
     // Initialize with new mnemonic
     const breezApiKey = Constants.expoConfig?.extra?.BREEZ_API_KEY
@@ -42,7 +42,7 @@ export async function restoreWallet(store: IWalletStore, mnemonic: string) {
       mnemonic: mnemonic,
     })
 
-    store.isInitialized = true
+    store.setInitialized(true)
     store.setError(null)
 
     // Fetch initial balance

--- a/app/models/wallet/actions/restoreWallet.ts
+++ b/app/models/wallet/actions/restoreWallet.ts
@@ -17,7 +17,7 @@ export async function restoreWallet(store: IWalletStore, mnemonic: string) {
     store.setPendingSendSat(0)
     store.setPendingReceiveSat(0)
     store.setTransactions([])
-    store.setMnemonic(null)
+    store.setMnemonic(undefined)  // Changed from null to undefined
 
     // Validate and save mnemonic to secure storage
     const saved = await SecureStorageService.setMnemonic(mnemonic)

--- a/app/models/wallet/types.ts
+++ b/app/models/wallet/types.ts
@@ -5,11 +5,11 @@ export interface IWalletStoreBase extends IStateTreeNode {
   isInitialized: boolean
   setupComplete: boolean
   error: string | null
-  mnemonic: string | null
+  mnemonic: string | undefined
   setBalanceSat: (balanceSat: number) => void
   setPendingReceiveSat: (pendingReceiveSat: number) => void
   setPendingSendSat: (pendingSendSat: number) => void
-  setMnemonic: (mnemonic: string) => void
+  setMnemonic: (mnemonic: string | undefined) => void
   setError: (message: string | null) => void
   setInitialized: (isInitialized: boolean) => void
   setSetupComplete: (complete: boolean) => void

